File: python/functions/sum_of_digits.py
# Function to calculate the sum of digits of a number

def sum_of_digits(n):
    # Handle negative numbers
    n = abs(n)

    total = 0

    # Extract digits one by one
    while n > 0:
        digit = n % 10       # Get last digit
        total += digit       # Add digit to total
        n //= 10             # Remove last digit

    return total


# Example input
number = 1234

# Print the result
print(sum_of_digits(number))

# Output:
# 10

File: C/sumofdigits.c
#include <stdio.h>

int sum_of_digits(int n)
{
    int sum = 0;

    if (n < 0)
        n = -n;

    while (n > 0)
    {
        sum += n % 10;
        n /= 10;
    }

    return sum;
}

int main()
{
    int n;
    printf("Enter the number: ");
    scanf("%d", &n);

    printf("The sum of digits of %d = %d\n", n, sum_of_digits(n));
    return 0;
}
